src/main/java/com/lob/client/AsyncLobClient.java

```diff
@@ -532,12 +532,18 @@ public void onThrowable(final Throwable t) {
         public T onCompleted(final Response response) throws Exception {
             if (isSuccess(response)) {
                 final T value = MAPPER.readValue(response.getResponseBody(), clazz);
+
+                ((AbstractResponse) value).setStatusCode(response.getStatusCode());
+                ((AbstractResponse) value).setHeaders(response.getHeaders());
+
                 guavaFut.set(value);
                 return value;
             }
             else {
                 final ErrorResponse error = MAPPER.readValue(response.getResponseBody(), ErrorResponse.class);
                 final LobApiException exception = new LobApiException(response.getUri(), error);
+                exception.setStatusCode(response.getStatusCode());
```

Comment:
[`response.getStatusCode()`](https://asynchttpclient.github.io/async-http-client/apidocs/com/ning/http/client/Response.html#getStatusCode()) returns an `int` so I think it's safe to store it as such.